addons/web/tests/test_image.py

```diff
@@ -42,3 +42,20 @@ def test_01_content_image_resize_placeholder(self):
         response = self.url_open('/web/image/fake/0/image_no_size')
         image = Image.open(io.BytesIO(response.content))
         self.assertEqual(image.size, (256, 256))
+
+    def test_02_content_image_Etag_304(self):
+        """ """
+
+        attachment = self.env['ir.attachment'].create({
+            'datas': b"R0lGODdhAQABAIAAAP///////ywAAAAAAQABAAACAkQBADs=",
+            'name': 'testEtag.gif',
+            'public': True,
+            'datas_fname': 'testEtag.gif',
+            'mimetype': 'image/gif',
+        })
+        response = self.url_open('/web/image/%s' % attachment.id)
+        self.assertEqual(response.status_code, 200)
+        etag = response.headers.get('Etag')
+
+        response2 = self.url_open('/web/image/%s' % attachment.id, headers={"If-None-Match": etag})
+        self.assertEqual(response2.status_code, 304)
```

Comment:
Is it possible to also test the content is returned correctly on the first request, and not here?

Reply:
Yes good idea